autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/Buttons/SaveScheduleButton.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { IconButton, Tooltip } from '@material-ui/core';
+import LockIcon from '@material-ui/icons/Lock';
+import LockOpenIcon from '@material-ui/icons/LockOpen';
+import { saveSchedule, unsaveSchedule } from '../../../../redux/actions/schedules';
+import { containsSchedule } from '../../../../redux/reducers/schedules';
+import { RootState } from '../../../../redux/reducer';
+import Meeting from '../../../../types/Meeting';
+import SmallFastProgress from '../../../SmallFastProgress';
+import * as styles from '../SchedulePreview.css';
+
+interface SaveScheduleButtonProps {
+  index: number;
+}
+
+const SaveScheduleButton: React.FC<SaveScheduleButtonProps> = ({ index }) => {
+  const dispatch = useDispatch();
+  const allSchedules = useSelector<RootState, Meeting[][]>((state) => state.schedules.allSchedules);
+  const savedSchedules = useSelector<RootState, Meeting[][]>((state) => (
+    state.schedules.savedSchedules
+  ));
+  const [saved, setSaved] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+
+  // schedule indices may change when schedules is changed, so refresh state whenever this happens
+  // or whenever saved schedules change
+  React.useEffect(() => {
+    setSaved(containsSchedule(savedSchedules, allSchedules[index]));
+  }, [allSchedules, savedSchedules, index]);
+
+  function sleep(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  // TODO: Once API for saving schedules is created, call it here
+  const handleClick = (): void => {
+    // do nothing if loading
+    if (loading) return;
+
+    setLoading(true);
+
+    if (saved) {
+      dispatch(unsaveSchedule(index));
+    } else {
+      dispatch(saveSchedule(index));
+    }
+
+    sleep(1000).then(() => setLoading(false));
+  };
+
+  // change icon and background color based on whether schedule is saved or not
+  let icon = saved ? <LockIcon /> : <LockOpenIcon />;
+  let tooltipText = saved ? 'Unsave' : 'Save';
+
+  if (loading) {
+    icon = <SmallFastProgress size="small" />;
+    tooltipText = saved ? 'Saving...' : 'Unsaving...';
+  }
+
+  return (
+    <Tooltip title={tooltipText} placement="top">
+      <IconButton className={styles.scheduleButton} size="small" onClick={handleClick} data-testid="save-schedule">
```

Comment:
Ik `small` is the smallest size MUI provides, but I think the lock & delete icons are still a bit too large - I think they should be shorter than or equal to the schedule text itself

Reply:
So apparently icons have a weird behavior where the icon size isn't actually changed by putting `size="small"` on the `IconButton` unless you also put `size="inherit"` on the icon itself, which I didn't notice until you pointed this out. Here's what it looks like now:
![image](https://user-images.githubusercontent.com/28655462/87224909-baf97f00-c34e-11ea-95a8-dafe03ebc427.png)
And here's what it looks like with `size="inherit"` on the buttons:
![image](https://user-images.githubusercontent.com/28655462/87224947-de242e80-c34e-11ea-9e68-02605e431145.png)
Let me know which you like better, or any suggestions for other options